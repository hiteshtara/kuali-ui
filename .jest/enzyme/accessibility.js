/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
