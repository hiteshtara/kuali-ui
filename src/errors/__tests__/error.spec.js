/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import Error from '..'

describe('Error Component', () => {
  it('should render an Error component with defaults', () => {
    const wrapper = shallow(<Error />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render an Error component with custom props', () => {
    const wrapper = shallow(<Error title='Test title' message='Test message' />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render an Error component with custom icon', () => {
    const wrapper = shallow(<Error customIconClass='fa fa-warning' />)
    expect(wrapper).toMatchSnapshot()
  })
})
