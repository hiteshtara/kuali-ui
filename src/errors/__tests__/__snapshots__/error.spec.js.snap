// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component should render an Error component with custom icon 1`] = `
<div
  className="error__container"
>
  <i
    aria-hidden="true"
    className="error__icon fa fa-warning"
  />
  <div
    className="error__title"
  >
    UH OH! Something went wrong!
  </div>
  <div
    className="error__message"
  >
    <p
      key="0"
    >
      Sorry for the inconvenience, it looks like we just had a server error.
    </p>
    <p
      key="1"
    >
          Our team of developers has been notified and will get this problem fixed ASAP!
    </p>
  </div>
</div>
`;

exports[`Error Component should render an Error component with custom props 1`] = `
<div
  className="error__container"
>
  <Icon
    aria-hidden={true}
    className="error__icon"
    iconClassName="material-icons"
    name="warning"
    variant="default"
  />
  <div
    className="error__title"
  >
    Test title
  </div>
  <div
    className="error__message"
  >
    <p
      key="0"
    >
      Test message
    </p>
  </div>
</div>
`;

exports[`Error Component should render an Error component with defaults 1`] = `
<div
  className="error__container"
>
  <Icon
    aria-hidden={true}
    className="error__icon"
    iconClassName="material-icons"
    name="warning"
    variant="default"
  />
  <div
    className="error__title"
  >
    UH OH! Something went wrong!
  </div>
  <div
    className="error__message"
  >
    <p
      key="0"
    >
      Sorry for the inconvenience, it looks like we just had a server error.
    </p>
    <p
      key="1"
    >
          Our team of developers has been notified and will get this problem fixed ASAP!
    </p>
  </div>
</div>
`;
